### DISCUSSION TITLE
How To Customize Error Handler Response

### DISCUSSION THREAD

**Discussion by hasanparasteh** [clue/framework-x#249] [Ideas]
Currently X framework always return an HTML error for 500, 405 and etc. I think it's a good idea to create an API in order to change the behavior and customize it. for example we want to return `json` for every 500, 405 and etc errors.

schema would be like:

```json
{
    "result": false,
    "message": "error message goes here"
}
```

There's no documents about these kind of configurations and I didn't find it anywhere

**Comment by SimonFrings** [comment]
Hey @hasanparasteh, thanks for bringing this up :+1:

We already talked about this while looking for features for our upcoming release, see https://twitter.com/x_framework/status/1762499173830778907. There also was a similar pull request in #238 about the same topic. This is definitely on our roadmap and will probably be a feature addition in the near future! While it might not make it into today's version, we're confident it will be included in an upcoming release.

To keep you in the loop, once the first release of Framework X is out, we quickly want to work on refactoring the current handler structure, reworking some existing handlers, and adding new ones. We'll probably address this topic along the way or shortly afterwards.

I'll keep this ticket updated as we make progress :+1:

If you need this urgently, you can also reach out to us or help us prioritizing by [becoming as sponsor](https://github.com/sponsors/reactphp) ❤️

FYI: I converted this ticket to a discussion as I think it's better suited for our "Ideas" section. We recommend opening bug tickets only when you can provide evidence of something being broken, just as a friendly reminder for future tickets :)

**Reply by hasanparasteh** [reply]
Thank you for your consideration @SimonFrings. 
sorry I keep that in mind and I'll open discussions for ideas.
